product_code/lagom/course_service/api/src/main/scala/de/upb/cs/uc4/course/model/Course.scala

```diff
@@ -53,39 +39,42 @@ case class Course(
-    val descriptionRegex = """[\s\S]*""".r // Allowed characters  for description
-    val dateRegex = """(\d\d\d\d)-(\d\d)-(\d\d)""".r
+    val nameRegex = """[\s\S]{1,100}""".r // Allowed characters for coursename "[a-zA-Z0-9\\s]+".r
+    val descriptionRegex = """[\s\S]{0,10000}""".r // Allowed characters  for description
+    val dateRegex = """^(?:(?:(?:(?:(?:[1-9]\d)(?:0[48]|[2468][048]|[13579][26])|(?:(?:[2468][048]|[13579][26])00))(-)(?:0?2\1(?:29)))|(?:(?:[1-9]\d{3})(-)(?:(?:(?:0?[13578]|1[02])\2(?:31))|(?:(?:0?[13-9]|1[0-2])\2(?:29|30))|(?:(?:0?[1-9])|(?:1[0-2]))\2(?:0?[1-9]|1\d|2[0-8])))))$""".r
+
 
     var errors = List[SimpleError]()
 
-    if (course.courseName == "") {
-      errors :+= SimpleError("courseName", "Course name must not be empty.")
+
+    courseName match {
+      case "" => errors :+= SimpleError("courseName", "Course name must not be empty.")
+      case _ if(!(nameRegex.matches(courseName))) => errors :+= SimpleError("courseName", "Course name must contain 1 to 100 characters.")
+      case _ =>
     }
-    if (!(nameRegex.matches(course.courseName))) {
-      errors :+= SimpleError("courseName", "Course name must only contain [..].")
+    if (!CourseType.All.map(_.toString).contains(courseType)) {
+      errors :+= SimpleError("courseType", "Course type must be one of " + CourseType.All.map(_.toString).reduce((a,b) => a+", "+b) + ".")
     }
-    if (!CourseType.All.contains(course.courseType)) {
-      errors :+= (SimpleError("courseType", "Course type must be one of [Lecture, Seminar, ProjectGroup]."))
+    if (!dateRegex.matches(startDate)) {
+      errors :+= SimpleError("startDate", "Start date must be of the following format \"yyyy-mm-dd\".")
     }
-    if (!dateRegex.matches(course.startDate)) {
-      errors :+= (SimpleError("startDate", "Start date must be of the following format \"yyyy-mm-dd\"."))
+    if (!dateRegex.matches(endDate)) {
+      errors :+= SimpleError("endDate", "End date must be of the following format \"yyyy-mm-dd\".")
     }
-    if (!dateRegex.matches(course.endDate)) {
-      errors :+= (SimpleError("endDate", "End date must be of the following format \"yyyy-mm-dd\"."))
+    if (ects <= 0 || ects > 1000) {
+      errors :+= SimpleError("ects", "ECTS must be a positive integer between 1 and 999.")
     }
-    if (course.ects <= 0) {
-      errors :+= (SimpleError("ects", "Ects must be a positive integer."))
+    if (maxParticipants <= 0 || maxParticipants > 10000) {
+      errors :+= SimpleError("maxParticipants", "Maximum participants must be a positive integer between 1 and 9999.")
     }
-    if (course.maxParticipants <= 0) {
-      errors :+= (SimpleError("maxParticipants", "Maximum Participants must be a positive integer."))
+    if (currentParticipants < 0 || currentParticipants > maxParticipants) {
+      errors :+= SimpleError("currentParticipants", "Current participants must be a positive integer between 0 and maximum number of participants.")
```

Comment:
Will we ever see this error? We do not change the number of current participants, this would be done in a later sprint, when students enroll to courses. Still wouldn't hurt to keep it in here of course.
```suggestion
      errors :+= SimpleError("currentParticipants", "Number of current participants must be a positive integer between 0 and maximum number of participants.")
```